lib/compress/huf_compress.c

```diff
@@ -1244,6 +1235,51 @@ typedef struct {
 #define SUSPECT_INCOMPRESSIBLE_SAMPLE_SIZE 4096
 #define SUSPECT_INCOMPRESSIBLE_SAMPLE_RATIO 10  /* Must be >= 2 */
 
+unsigned HUF_minTableLog(size_t srcSize, unsigned maxSymbolValue)
+{
+    U32 minBitsSrc = ZSTD_highbit32((U32)(srcSize)) + 1;
+    U32 minBitsSymbols = ZSTD_highbit32(maxSymbolValue) + 2;
```

Comment:
#### The little detail that matters: 

Note that, in this formula, `+ 2` is itself an heuristic.
That's because this code is taken from `FSE_minTableLog()`, where it's presumed to be employed in combination with a fast heuristic.
Consequently, one of the goals was to avoid providing a too low minimal, that would end up being, in general, a rather poor candidate.

But now, with this new brute-force strategy, as long as the candidate is sometimes a better one, it's worth investigating.

So it's possible to change this formula by `+ 1`, which is the real minimal.
(Note : the real minimal should actually be based on the cardinality of the distribution, for which `maxSymbolValue` is merely a cheap upper bound)

I tried that, with a quick test on `silesia.tar`.
The new low limit allows finding a few more bytes here and there, achieving an additional 1-2 KB savings at higher compression modes. Not big in absolute value, but relative to existing savings, it improves this strategy by ~20%, so that's a non-negligible contributor.

Where it shines though is in combination with small blocks.
For example, when cutting `silesia.tar` into blocks of 1 KB, it achieves additional savings of almost 300 KB. Not a mistake. So it's a game changer in this case.

The downside of this strategy is that there is now one more distribution to test.
So it's even slower.
That could imply revisiting the algorithm's triggering threshold.

Alternatively, it could also provide an incentive to invest time into a more optimized method, that requires less cpu effort. Which, I believe, is possible, without loss of compression ratio. Maybe this could be dealt with in a follow up.